src/test/java/uk/gov/pay/directdebit/payments/api/PaymentViewValidatorTest.java

```diff
@@ -16,32 +22,109 @@
-        PaymentViewSearchParams searchParams = new PaymentViewSearchParams("a-gateway-account", null, null);
-        validator.validateParams(searchParams);
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams("a-gateway-account", null, null, null, null);
+        searchParams = validator.validateParams(searchParams);
         assertThat(searchParams.getPaginationParams().getDisplaySize(), is(500l));
         assertThat(searchParams.getPaginationParams().getPageNumber(), is(1l));
     }
+    
+    @Test
+    public void shouldCorrectlyValidate_toAndFromDate() {
+        String fromDate = "2018-05-03T15:00Z";
+        String toDate = "2018-05-04T15:00Z";
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams("a-gateway-account",
+                3l, 
+                256l,
+                fromDate,
+                toDate);
+        searchParams = validator.validateParams(searchParams);
+        assertThat(searchParams.getSearchDateParams().getFromDate().toString(), is(fromDate));
+        assertThat(searchParams.getSearchDateParams().getToDate().toString(), is(toDate));
+    }
+    
+    @Test 
+    public void shouldLeaveToDateNull_whenMissing() {
+        String fromDate = "2018-05-03T15:00Z";
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams("a-gateway-account",
+                3l,
+                256l,
+                fromDate,
+                null);
+        searchParams = validator.validateParams(searchParams);
+        assertThat(searchParams.getSearchDateParams().getFromDate().toString(), is(fromDate));
+        assertThat(searchParams.getSearchDateParams().getToDate(), is(nullValue()));
+    }
+    
+    @Test
+    public void shouldLeaveFromDateNull_whenMissing() {
+        String toDate = "2018-05-04T15:00Z";
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams("a-gateway-account",
+                3l,
+                256l,
+                null,
+                toDate);
+        searchParams = validator.validateParams(searchParams);
+        assertThat(searchParams.getSearchDateParams().getFromDate(), is(nullValue()));
+        assertThat(searchParams.getSearchDateParams().getToDate().toString(), is(toDate));
+    }
+    
+    @Test
+    public void shouldThrowInvalidDateException_whenFromDateIsAfterToDate() {
+        String fromDate = "2018-05-05T15:00Z";
+        String toDate = "2018-05-04T15:00Z";
+        thrown.expect(InvalidDateException.class);
+        thrown.expectMessage("from_date (2018-05-05T15:00Z) must be earlier then to_date (2018-05-04T15:00Z)");
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams("a-gateway-account",
+                3l,
+                256l,
+                fromDate,
+                toDate);
+        validator.validateParams(searchParams);
+    }
+
+    @Test
+    public void shouldThrowUnparsableDateException_whenFromDateIsMailFormed() {
```

Comment:
Is it ok to add some `SQL` related characters when we submit valid date as a `String` but append some characters like `'` , `%`, etc.. just in case?

Reply:
👍